plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/LinuxExec.kt

```diff
@@ -98,32 +98,58 @@ open class LinuxExec @Inject constructor(objects: ObjectFactory) : ConclaveTask(
     }
 
     /** Returns the ERROR output of the command only, in the returned list. */
-    fun exec(params: List<String>): List<String>? {
+    fun exec(params: List<String>, dockerWorkdirPath: String? = null, throwsException: Boolean=false): List<String>? {
         val errorOut = ByteArrayOutputStream()
-        val args: List<String> = if (buildInDocker.get()) getDockerRunArgs(params) else params
+        val args: List<String> = if (buildInDocker.get()) getDockerRunArgs(params, dockerWorkdirPath) else params
 
         val result = commandLine(
             args,
             commandLineConfig = CommandLineConfig(ignoreExitValue = true, errorOutputStream = errorOut)
         )
 
-        if (result.exitValue == 137) {
+        if (result.exitValue != 0) {
+            handleError(result.exitValue, errorOut, throwsException)
+        }
+        result.assertNormalExitValue()
+        return null
+    }
+
+    private fun handleError(exitCode:Int, errorOut :ByteArrayOutputStream, throwsException: Boolean): List<String> {
+        if (exitCode == 137) {
             // 137 = 128 + SIGKILL, which happens when the kernel out-of-memory killer runs.
             throwOutOfMemoryException()
-        }
-        if (result.exitValue != 0) {
+        } else {
             errorOut.writeTo(System.err)
             // Using default charset here because the strings come from a sub-process and that's what they'll pick up.
             // Hopefully it's UTF-8 - it should be!
-            return String(errorOut.toByteArray()).split(System.lineSeparator())
+            val errorString = String(errorOut.toByteArray())
+
+            if (throwsException) {
+                throw GradleException(errorString)
+            } else {
+                return errorString.split(System.lineSeparator())
+            }
         }
-        result.assertNormalExitValue()
-        return null
     }
 
-    /** Returns the output of the command executed in the container. */
-    fun execWithOutput(params: List<String>): String {
-        return commandWithOutput(*getDockerRunArgs(params).toTypedArray())
+    fun execWithOutput(params: List<String>, dockerWorkdirPath: String? = null, throwsException: Boolean=false): String {
```

Comment:
We seem to be duplicating a lot of command execution code. Why does this exist?

Reply:
I think we need to refactor the "normal" LinuxExec.exec`. This does not give the possibility of getting back the output when the command succeeds and does not throw exceptions when failing.